ssc/Pos/Ssc/Logic/VAR.hs

```diff
@@ -224,7 +224,10 @@ sscRollbackU
 sscRollbackU blocks = tossToUpdate $ rollbackSsc oldestEOS payloads
   where
     oldestEOS = blocks ^. _Wrapped . _neLast . epochOrSlotG
-    payloads = over _Wrapped (map snd . rights . toList) blocks
+    payloads = NewestFirst $ map bcmPayload $ toList blocks
+    -- payloads = case toList blocks of
+    --     [ComponentBlockMain _ a]  -> NewestFirst $ map bcmPayload $ toList blocks
+    --     [ComponentBlockGenesis _] -> NewestFirst $ _ -- ???
 
```

Comment:
@gromakovsky this compiles but I know I need to pattern match in order not to blow on runtime. What is appropriate way to handle second case ? Is there a way to construct `SscPayload` from thin air ?

Reply:
You can write something like (I didn't type check it)
```
    payloads = NewestFirst $ mapMaybe extractPayload $ toList blocks
    extractPayload (ComponentBlockMain _ a) = Just a
    extractPayload (ComponentBlockGenesis _) = Nothing
```

Reply:
It compiles, thanks